### PR TITLE
Rebuild model if necessary, fixes #7308

### LINKED ISSUE #7308: model.get_layer fails to work if called a second time after adding a new layer
model.get_layer() does not correctly return the correct layer if a new layer is added after it is first used.

To replicate (using keras 2.0.6)

from keras.models import Sequential
from keras.layers import Dense, Dropout, Conv2D, Flatten, Reshape

model = Sequential()
model.add(Dense(128, input_shape=(784,)))
model.add(Dense(64))
model.get_layer(index=-1).output_shape
model.add(Dense(32))
model.get_layer(index=-1).output_shape

----- outputs: (None, 64), rather than the expected (None, 32)

### PR DESCRIPTION
**Pull request by Danielhiversen**

### CODE REVIEW

**Comment by Dref360** [comment]
Need a test (Same code as the issue.)

**Comment by Danielhiversen** [comment]
Fixed
